### ISSUE TITLE
Ability to delete rooms.

### ISSUE THREAD

**Issue by ara4n** [matrix-org/matrix-doc#1882]
A fairly common request is the ability to "delete a room".

In practice, this would mean someone with sufficient power marking with room with a tombstone with metadata requesting all participating servers drop the room from their DB.  However, this would vandalise everyone else's copy of the room.

Perhaps a better bet would be to support room garbage collection, such that if everyone on your local server has hit 'leave' and 'forget' on their copy of the room, the room is actually removed from the server.

**Comment by turt2live** [comment]
https://matrix.org/docs/spec/client_server/unstable.html#post-matrix-client-r0-rooms-roomid-forget already says:

> If all users on a homeserver forget a room, the room is eligible for deletion from that homeserver.

so this sounds like an implementation detail which synapse doesn't support?

**Comment by ara4n** [comment]
true. i guess i'm more trying to work out what use case something like https://twitter.com/cartocalypse/status/1095322661322977281 is asking for.

**Comment by turt2live** [comment]
I suppose coming from a UI perspective we don't really have the endpoint for this, in which case a dedicated `/delete` enpoint makes sense (using the behaviour you've described).

**Comment by dud1337** [comment]
Using a synapse for a workplace server, this would make usability significantly easier. In particular for short-term rooms, a room administrator should be able to close a room. Equivalent to kicking everyone and waiting for garbage collection.

**Comment by LewisDodgson** [comment]
Any update on this? Seems like it would be a fairly common thing to do.

**Comment by hannojg** [comment]
There is this PR: https://github.com/matrix-org/synapse/pull/7613

**Comment by ell1e** [comment]
@hannojg this is just for server admins though, it would be nice if this could be extended to room admins as well (so e.g. random people creating personal rooms on `matrix.org` can remove them again if they want to).

**Comment by rugk** [comment]
See

Possibbility to clear whole chat to remove all messages / destroy chat room
in https://github.com/vector-im/riot-web/issues/14401

If you want, I can copy and recreate this issue here?

**Comment by ell1e** [comment]
I mean there is a difference for private chats and public rooms. That issue is mostly concerned with private chats I think, I am thinking of public rooms I created but then all room admins decide the room is no longer to be kept (e.g. because it moved elsewhere) and want to actually delete it, beyond just removing the messages.

Basically, history clearing seems related, but not necessarily the same thing. Although I guess both could be solved by allowing the room owners (!= just server admins) to delete a room entirely with all its history and contents.

**Comment by rugk** [comment]
Fine, created https://github.com/matrix-org/matrix-doc/issues/2686.

**Comment by richvdh** [comment]
> @hannojg this is just for server admins though, it would be nice if this could be extended to room admins as well (so e.g. random people creating personal rooms on `matrix.org` can remove them again if they want to).

For clarity: when @ell1e said "this is just for server admins", I think they were referring to matrix-org/synapse#7613. *That* PR adds an API for use by server admins. *This* issue (#1882) covers the usecase of room admins deciding that a room should be deleted.

**Comment by ell1e** [comment]
@richvdh sorry, yes. that is exactly what I meant, I apologize for my unclear use of "this" that was somewhat confusing in retrospect

**Comment by silvernode** [comment]
I wanted to delete my room and recreate it but I can't, so now the new room has to have a different alias. This is unfortunate since I already left the original room, hoping that would delete it if nobody else was present and now I can't get back in to change the alias in order to give it to my new room.

**Comment by TheEvilSkeleton** [comment]
@silvernode contacting the homeserver maintainers and asking them if they can remove the room should work. That's how we did.

**Comment by silvernode** [comment]
> @silvernode contacting the homeserver maintainers and asking them if they can remove the room should work. That's how we did.

I can try sending an email to matrix.org i suppose. Thanks for the reply.

**Comment by guglovich** [comment]
Relevant. While I was figuring out what was what, I already had several rooms of corpses with occupied names. It's too bad that such a basic function as removal was not implemented.
